BinnedCountPredictionModel.py

```diff
@@ -0,0 +1,468 @@
+'''
+Purpose of this Resnet is to predict binned amount of lightning counts based on ABI data. For instance, one may set the bins = [10,50,100]. 
+This will set the bins for the predictions equal to [0,1,10,50,100], as the code is always set to include 0 and 1 as bins for the lightning counts. 
+Code will also output a plot .png 'BinConfusion.png' of the confusion matrix for the results of the code
+Will also output the best model as best.pt 
+'''
+
+import random
+import pickle
+import numpy as np
+
+from torch.nn.modules.loss import _WeightedLoss
+from torch.optim.lr_scheduler import *
+from torch.autograd import Variable
+import torch.nn.functional as F
+import torch.nn as nn
+import torch
+
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import StandardScaler
+from sklearn import metrics
+import torchvision.models as models
+
+
+from tqdm import tqdm as tqdm_base
+#pull arguments
+
+from tqdm import tqdm as tqdm_base
+#get cuda instance
+def tqdm(*args, **kwargs):
+    if hasattr(tqdm_base, '_instances'):
+        for instance in list(tqdm_base._instances):
+            tqdm_base._decr_instances(instance)
+    return tqdm_base(*args, **kwargs)
+
+is_cuda = torch.cuda.is_available()
+device = torch.device(torch.cuda.current_device()) if is_cuda else torch.device("cuda")
+
+if is_cuda:
+    torch.backends.cudnn.benchmark = True
+
+print(f'Preparing to use device {device}')
+#Download Data
+with open('/glade/u/home/gwallach/goes16ci/train_data_scaled.pkl','rb') as f:
+    x = pickle.load(f)
+    
+print("Saved shape:", x.shape)
+x = x.transpose(0,3,1,2)
+print("Reshaped to:", x.shape)
+
+with open('/glade/u/home/gwallach/goes16ci/train_counts.pkl','rb') as f:
+    y = pickle.load(f)
+    
+
+print("Saved shape:", y.shape)
+y = y.reshape(y.shape[0], 1)
+print("Reshaped to:", y.shape)
+#split into test and train
+X_train, X_test, Y_train, Y_test = train_test_split(
+    x, y, test_size=0.2, random_state = 5000
+)
+
+y_scaler = StandardScaler()
+#split data into bins
+bins = [5,10,15,25,50,75,100]
+
+y_train = np.where(Y_train[:] > 0.0, 1, 0)
+y_test = np.where(Y_test[:] > 0.0, 1, 0)
+for p,q in enumerate(bins):
+    y_train = np.where(Y_train[:] > q, p+2, y_train)
+    y_test = np.where(Y_test[:] > q, p+2, y_test)
+    
+print(y_train.shape)
+print(y_test.shape)
+print(X_train.shape)
+print(X_test.shape)
+##create weights based on the counts
+from collections import Counter
+counts = Counter()
+for val in y_train:
+    counts[val[0]] += 1
+counts = dict(counts)
+
+#weights = [1 - (counts[x] / sum(counts.values())) for x in sorted(counts.keys())]
+weights = [np.log1p(max(counts.values()) / counts[x]) for x in sorted(counts.keys())]
+weights = [x / max(weights) for x in weights]
+weights = torch.FloatTensor(weights).to(device)
+
+print(weights)
+##Load a Model
+class ResNet(nn.Module):
```

Comment:
Classes defined in this module should be in goes16ci/models.py rather than in a script. That way other training scripts can import and use the model.